### PR TITLE
RavenDB-10732: LowMem use smaps instead of stats for available memory…

### PR DESCRIPTION
**Pull request by aviviadi**
… calc

### CODE REVIEW

File: src/Sparrow/Platform/Posix/SmapsReader.cs [hunk 1]
```diff
@@ -7,6 +7,81 @@
...
+
+        public DynamicJsonArray ReturnResults()
+        {
+            return _dja;
+        }
+    }
+    
+    public struct SmapsReaderNoAllocResults : ISmapsReaderResultAction
+    {
+        private SmapsReaderResults[] _results;
```
Comment (ayende) [inline src/Sparrow/Platform/Posix/SmapsReader.cs]: Use `List<T>` here, instead. No need to write such code ourselfs.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Sparrow/Platform/Posix/SmapsReader.cs (hunk 1 above)
Comment (aviviadi) [reply src/Sparrow/Platform/Posix/SmapsReader.cs]: Thought about lowering allocations to minimum

Comment (ayende) [reply src/Sparrow/Platform/Posix/SmapsReader.cs]: You aren't, you still do `resize` in the end, and you don't need to, and this is how `List` already work

Comment (aviviadi) [reply src/Sparrow/Platform/Posix/SmapsReader.cs]: and we do not use these results anyhow.  maybe just throw not implemented ?

Comment (ayende) [reply src/Sparrow/Platform/Posix/SmapsReader.cs]: why throw? Why do _anything_ ? Just have empty method.

**Pull request merged**